drivers/sensor/mpu6050/mpu6050.c

```diff
@@ -40,11 +40,28 @@ static void mpu6050_convert_gyro(struct sensor_value *val, int16_t raw_val,
 }
 
 /* see "Temperature Measurement" section from register map description */
-static inline void mpu6050_convert_temp(struct sensor_value *val,
+static inline void mpu6050_convert_temp(enum mpu6050_device_type device_type,
+					struct sensor_value *val,
 					int16_t raw_val)
 {
-	val->val1 = raw_val / 340 + 36;
-	val->val2 = ((int64_t)(raw_val % 340) * 1000000) / 340 + 530000;
+	int64_t tmp_val = (int64_t)raw_val * 1000000;
+
+	switch (device_type) {
+	case DEVICE_TYPE_MPU6500:
+	{
+		tmp_val = (tmp_val / 334) + 21000000;
+		break;
+	}
+
+	case DEVICE_TYPE_MPU6050:
+	default:
+	{
+		tmp_val = (tmp_val / 340) + 36000000;
+	}
+	};
+
+	val->val1 = tmp_val / 1000000;
+	val->val2 = tmp_val % 1000000;
 
 	if (val->val2 < 0) {
```

Comment:
This bit (lines 66-71) should no longer be needed now you are using the intermediate value for calculation.

Reply:
They are needed for negative values, e.g. -0.3c